### PR TITLE
feat(new): Azure Flexible PSQL

### LINKED ISSUE #302: [Module Proposal]: Azure Database for PostgreSQL - Single Server
### Have you checked this module does not already exist in the registry?

Yes

### Why is the module needed?

While we are aware that Postgresql Single Server is currently [on the retirement path](https://learn.microsoft.com/en-us/azure/postgresql/single-server/whats-happening-to-postgresql-single-server), Postgresql Flexible Server currently has no support for Private Endpoints which is a requirement for us.

Therefore, we'll need a Bicep module for this.

### Module path

database/postgresql-single-server

### Describe the module

This module will allow to create [PostgreSQL Single Server ](https://learn.microsoft.com/en-us/azure/postgresql/single-server/overview-single-server) resource.

### PR DESCRIPTION
**Pull request by dciborow**
Closes #302

## Description

If you haven't already, read the full [contribution guide](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, edit the PR description and run through the relevant checklist below.

Enable GitHub Worksflows in your fork to enable auto-generation of assets with our [GitHub Action](/.github/workflows/push-auto-generate.yml).
To trigger GitHub Actions after auto-generation, [add a GitHub PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) as a secret in your forked repository called `PAT`.

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [x] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

### CODE REVIEW

**Comment by shenglol** [comment]
Maybe making `flexible` a common prefix (`storage/flexible-postgresql-server`) is better - I believe there will a need for `storage/flexible-myseql-server`.

**Comment by dciborow** [comment]
> Maybe making `flexible` a common prefix (`storage/flexible-postgresql-server`) is better - I believe there will a need for `storage/flexible-myseql-server`.

This should be ready for registry onboarding now.

File: modules/storage/flexible-postgresql-server/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+| :--- | :--: | :---------- |
+
+## Examples
+
+### Example 1
+
+This example creates a new PostgreSQL Server with a unique name in the East US region using the default PostgreSQL Server configuration settings. The module output is the ID of the created PostgreSQL Server, which can be used in other Azure resource deployments.
+
+```bicep
+module postgresqlServer 'br/public:databases/flexible-postgresql:0.0.1' = {
```
Comment (shenglol) [inline modules/storage/flexible-postgresql-server/README.md:44]: Should we update the module path in examples?

File: modules/storage/flexible-postgresql-server/main.bicep [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+@description('The geo-redundant backup setting')
+@allowed([ 'Enabled', 'Disabled' ])
+param geoRedundantBackup string = 'Disabled'
+
+@description('The high availability mode')
+@allowed([ 'Disabled', 'Enabled' ])
+param highAvailability string = 'Disabled'
+
+// @description('Enable Azure Active Directory based authentication')
```
Comment (shenglol) [inline modules/storage/flexible-postgresql-server/main.bicep:56]: Does the commented code need to be removed?

File: modules/storage/flexible-postgresql-server/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
+# Azure Flexible PostGres-SQL
```
Comment (shenglol) [inline modules/storage/flexible-postgresql-server/README.md:1]:
```suggestion
# Azure Flexible PostGres-SQL server
```

File: modules/storage/flexible-postgresql-server/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+This Bicep module creates a PostgreSQL Server with zone-redundancy, virtual network access.
```
Comment (shenglol) [inline modules/storage/flexible-postgresql-server/README.md:3]: Is the summary and description below still true? The vnet and zone-redundancy resources are commented out in `main.bicep`.

Comment (shenglol) [reply modules/storage/flexible-postgresql-server/README.md:3]: If we are not going to implement vnet support in the PR, it should not close https://github.com/Azure/bicep-registry-modules/issues/302.

**Comment by shenglol** [comment]
Closing for now. We can reopen it once the vnet support is implemented.

**Pull request closed without merging**